src/components/lib/dpdk/adapter/cos_dpdk_adapter.c

```diff
@@ -0,0 +1,194 @@
+#include <llprint.h>
+#include <pci.h>
+#include <io.h>
+#include <cos_kernel_api.h>
+#include <cos_defkernel_api.h>
+
+#include<rte_bus_pci.h>
+
+#include "cos_dpdk_adapter.h"
+
+COS_DECLARE_MODULE(e1000);
+COS_DECLARE_MODULE(mempool_ring)
+
+static struct pci_dev cos_pci_devices[PCI_DEVICE_MAX];
+static int pci_dev_nb = 0;
+
+#define PRINT_BUF_SZ 512
+
+int
+cos_printc(const char *fmt, va_list ap)
+{
+	size_t  ret;
+	char print_buffer[PRINT_BUF_SZ];
+
+	ret = vsnprintf(print_buffer, PRINT_BUF_SZ, fmt, ap);
+	cos_llprint(print_buffer, ret);
+
+	return ret;
+}
+
+int
+cos_printf(const char *fmt,...)
```

Comment:
> Are these prints both for debugging purposes? I thought RTE_LOG is still usable.

No, this is not for debug. This is for RTE_LOG. We need an interface for DPDK to print. We need a very obvious name rather than `printc` to know this is a interface for DPDK.